Framework/Core/include/Framework/AnalysisDataModel.h

```diff
@@ -400,6 +400,62 @@ DECLARE_SOA_TABLE(Run2V0s, "AOD", "RUN2V0", run2v0::BCId,
                   run2v0::BBFlag, run2v0::BGFlag);
 using Run2V0 = Run2V0s::iterator;
 
+// ---- MC tables ----
+
+namespace mcvtx
+{
+DECLARE_SOA_INDEX_COLUMN(BC, bc);
+DECLARE_SOA_COLUMN(GeneratorsID, generatorsID, short);
+DECLARE_SOA_COLUMN(X, x, float);
+DECLARE_SOA_COLUMN(Y, y, float);
+DECLARE_SOA_COLUMN(Z, z, float);
+DECLARE_SOA_COLUMN(T, t, float);
+DECLARE_SOA_COLUMN(Weight, weight, float);
+DECLARE_SOA_COLUMN(NProduced, nProduced, int);
+} // namespace mcvtx
+
+DECLARE_SOA_TABLE(McVtxs, "AOD", "MCVTX", o2::soa::Index<>, mcvtx::BCId,
+		  mcvtx::GeneratorsID,
+                  mcvtx::X, mcvtx::Y, mcvtx::Z, mcvtx::T, mcvtx::Weight,
+                  mcvtx::NProduced);
+using McVtx = McVtxs::iterator;
+
+namespace label
+{
+DECLARE_SOA_COLUMN(Label, label, int);
+DECLARE_SOA_COLUMN(LabelMask, labelMask, uint16_t);
+} // namespace label
+
+DECLARE_SOA_TABLE(Labels, "AOD", "LABEL", o2::soa::Index<>,
+		  label::Label, label::LabelMask);
+using Label = Labels::iterator;
+
+namespace mcparticle
+{
+DECLARE_SOA_INDEX_COLUMN(Collision, collision);
```

Comment:
When we produce MC only, how do we do the indexing when we have no Collision entry?
Where do MC particles go which have not produced a valid collision vertex in the reconstruction?

Reply:
Here I will replace collision by mcollision (mcvtx will become mccollision) to avoid misunderstanding. In MC we always have collision (primary vertex).

Reply:
Very good!